File: dart/7kyu/xo/bin/xo.dart
import "package:test/test.dart";

void main() {
  group('Fixed tests', () {
    test("Testing for xo", () {
      expect(XO('xo'), equals(true));
    });
    test("Testing for XO", () {
      expect(XO('XO'), equals(true));
    });
    test("Testing for xo0", () {
      expect(XO('xo0'), equals(true));
    });
    test("Testing for xxxoo", () {
      expect(XO('xxxoo'), equals(false));
    });
    test("Testing for xxOo", () {
      expect(XO("xxOo"), equals(true));
    });
    test("Testing for empty string", () {
      expect(XO(''), equals(true));
    });
    test("Testing for xxxxxoooxooo", () {
      expect(XO('xxxxxoooxooo'), equals(true));
    });
    test("Testing for xxxm", () {
      expect(XO("xxxm"), equals(false));
    });
    test("Testing for ooom", () {
      expect(XO("ooom"), equals(false));
    });
    test("Testing for Oo", () {
      expect(XO("Oo"), equals(false));
    });
    test("Testing for abcdefghijklmnopqrstuvwxyz", () {
      expect(XO('abcdefghijklmnopqrstuvwxyz'), equals(true));
    });
  });
}

// ignore: non_constant_identifier_names
bool XO(String str) {
  var strList = str.toLowerCase().split("");
  return strList.where((e) => e == 'o').length ==
      strList.where((e) => e == 'x').length;
}


// bool XO(String str) {
//   var s = str.toLowerCase();
//   return 'x'.allMatches(s).length == 'o'.allMatches(s).length;
// }

File: typescript/7kyu/xo/xo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.xo = void 0;
function xo(str) {
    return (str.match(/o/gi) || []).length === (str.match(/x/gi) || []).length;
}
exports.xo = xo;
// export function xo(str: string) {
//     return str.toLowerCase().split('x').length == str.toLowerCase().split('o').length;
// }
const chai_1 = require("chai");
describe("xo", function () {
    it("Basic Tests", function () {
        chai_1.assert.strictEqual(xo('xo'), true);
        chai_1.assert.strictEqual(xo("xxOo"), true);
        chai_1.assert.strictEqual(xo("xxxm"), false);
        chai_1.assert.strictEqual(xo("Oo"), false);
        chai_1.assert.strictEqual(xo("ooom"), false);
    });
});
